src/main/java/uk/gov/verifiablelog/merkletree/InMemory.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.verifiablelog.merkletree;
+import java.util.TreeMap;
+
+public class InMemory implements MemoizationStore {
+
+    private TreeMap<Integer, TreeMap<Integer, byte[]>> data;
```

Comment:
It'd be worth expressing why we went with `TreeMap<Integer, TreeMap<Integer, byte[]>>` rather than what I would have expected: `HashMap<Pair<Integer,Integer>,byte[]>`